src/xSTIR/cSTIR/include/sirf/STIR/stir_x.h

```diff
@@ -1105,11 +1105,28 @@ The actual algorithm is described in
 		}
 	};
 
-	class xSTIR_GeneralisedObjectiveFunction3DF :
-		public stir::GeneralisedObjectiveFunction < Image3DF > {
+	class xSTIR_GeneralisedObjectiveFunction3DF : public ObjectiveFunction3DF {
 	public:
+		void compute_gradient(const STIRImageData& id, int subset, STIRImageData& gd)
```

Comment:
add a doxygen string please.

By the way, does SIRF not use the same convention as STIR, where the output arg `gd` is placed first?

Reply:
done